### PR TITLE
Feature/svc 1159 send json logs to datadog

### PR DESCRIPTION
**Pull request by andela-sjames**
https://themuse.atlassian.net/browse/SVC-1159

PR: Updates `Muselog` allowing it to send logs from the web service to DataDog. 

New env variables Alert!

DATADOG_HOST (Datadog agent URI)
DATADOG_UDP_PORT which defaults to 10518.

### CODE REVIEW

File: muselog/datadog.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import json
+from logging.handlers import DatagramHandler
+
+class DataDogUdpHandler(DatagramHandler):
```
Comment (ian-axelrod) [inline muselog/datadog.py]: Needs extra newline above.

File: muselog/datadog.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    A handler class which writes logging records, in pickle format, to
+    a datagram socket.  The pickle which is sent is that of the LogRecord's
+    attribute dictionary (__dict__), so that the receiver does not need to
+    have the logging module installed in order to process the logging event.
+
+    To unpickle the record at the receiving end into a LogRecord, use the
+    makeLogRecord function.
+    """
+
+    def __init__(self, host, port):
```
Comment (ian-axelrod) [inline muselog/datadog.py]: Needs typehints

File: muselog/datadog.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        """
+        Initializes the handler with a specific host address and port.
+
+        :param host: Datadog UDP input host
+        :param port: Datadog UDP input port
+        """
+
+        DatagramHandler.__init__(self, host, port)
+
+    def send(self, s):
```
Comment (ian-axelrod) [inline muselog/datadog.py]: Needs typehints

File: muselog/datadog.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        when the network is busy - UDP does not guarantee delivery and
+        can deliver packets out of sequence.
+        """
+
+        if self.sock is None:
+            self.createSocket()
+
+        self.sock.sendto(bytes(s+"\n", "utf-8"), (self.host, self.port))
+
+    def makePickle(self, record):
```
Comment (ian-axelrod) [inline muselog/datadog.py]: Needs typehints

File: setup.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from setuptools import setup
 
-VERSION = "1.2.0"
+VERSION = "1.2.1"
```
Comment (ian-axelrod) [inline setup.py]: This is a minor version change, not a patch version change.

**Comment by andela-sjames** [comment]
Updated

File: muselog/datadog.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    def __init__(self, host: str, port: int):
+        """
+        Initializes the handler with a specific host address and port.
+
+        :param host: Datadog UDP input host
+        :param port: Datadog UDP input port
+        """
+
+        DatagramHandler.__init__(self, host, port)
```
Comment (ian-axelrod) [inline muselog/datadog.py]: This should be `super().__init__(host, port)`

File: README.md [hunk 7]
```diff
@@ -33,6 +33,12 @@ You must set following environment variables to enable Graylog integration.
 - GRAYLOG_TLS_PORT          :: Graylog server port that `tls` handler type sends GELF messages to. (Default: 12201).
 - GRAYLOG_TLS_TIMEOUT_SECS  :: Number of seconds to wait for TCP ack before abandoning message send.
 
+### Datadog integation
```
Comment (ian-axelrod) [inline README.md]: Typo: Datadog integation -> Datadog integration

**Comment by andela-sjames** [comment]
Updated

File: muselog/datadog.py [hunk 8]
```diff
@@ -23,7 +23,7 @@ def __init__(self, host: str, port: int):
         :param port: Datadog UDP input port
         """
 
-        DatagramHandler.__init__(self, host, port)
+        super().__init__(self, host, port)
```
Comment (ian-axelrod) [inline muselog/datadog.py]: Does not need the `self` argument.

**Comment by ian-axelrod** [comment]
Can you add some tests for this? The fact `super().__init__(self, ...)` worked for the datadog host setup suggests that there are no tests covering the functionality, as that call will not work.

File: muselog/datadog.py (hunk 8 above)
Comment (andela-sjames) [reply muselog/datadog.py]: My bad... I didn't test the call to `super` on the Datadog agent. I will modify and run against the agent as well with a test.

**Comment by andela-sjames** [comment]
Updated

File: tests/test_setup_logging.py [hunk 9]
```diff
@@ -47,4 +47,5 @@ def tearDown(self):
         self.handler.close()
 
     def test_datadog_handler_called(self):
+        self.logger.warning("Testing handler 1 2 3")
         self.assertEqual(True, self.handler.send.called)
```
Comment (ian-axelrod) [inline tests/test_setup_logging.py]: I think a more thorough test is in order. Perhaps checking that the arguments to send are what you expect?

Comment (andela-sjames) [reply tests/test_setup_logging.py]: The arguments here are the default log records as we are not modifying the content in any way. It's a `dict` that is generated by the logging module.

Comment (andela-sjames) [reply tests/test_setup_logging.py]: I'll check for that message is exactly what's used.

**Pull request merged**